interfaces/builtin/acrn.go

```diff
@@ -0,0 +1,58 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2021 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+const acrnSummary = `allows access to the acrn_hsm device`
+
+const acrnBaseDeclarationSlots = `
+  acrn:
+    allow-installation:
+      slot-snap-type:
+        - core
+    deny-auto-connection: true
+`
+
```

Comment:
Does this interface need to be super-privileged? It is not clear to me from the documentation whether interacting with the ACRN HSM would allow a snap to cause the hypervisor to make changes to the service VM itself, or does this only allow to control the user VMs? If this does allow to change the service VM from within the service VM then I think it should be super privileged. In that case the snap would need to also declare `deny-connection: true` in `acrnBaseDeclarationSlots`.

Reply:
Hi @alexmurray , not sure how to answer your question as I don't know what you mean by "making changes to the Service VM itself". That interface (via the character device) is used to control the lifecycle of User VMs, but there are a number of other operations that the `acrn-dm` will perform in addition to talking to the hypervisor via that character device.

Reply:
Yes I understand that the interface is designed to control the User VMs - I was also wondering if perhaps it (intentionally or not) also permitted to control the Service VM as well?

Reply:
Hi @alexmurray , I'm asking our engineers/architects to confirm that since I'm not 100% sure of the answer and I wouldn't want to mislead you.

Reply:
Through the HSM driver, all hypercalls can only control the user VMs. It should not impact the service VM itself. The resource of user VM can be controlled by service VM or hypervisor, but the service VM itself is only can be controlled by hypervisor.

Reply:
Ok it sounds like this does not need to be super-privileged then - thanks for the extra info.

Reply:
The super-privileged means root permission of the hsm device node(cdev), right?

Reply:
No, super-privileged means that to use a particular interface, a snap requires a manual declaration be issued by the administrator of the particular store where the snap is uploaded to. Some interfaces essentially allow a snap to take ownership of a device (e.g. the `snapd-control` or `docker-support` interfaces both grant so many permissions to a snap that it can administrer the device the snap is running on). I was trying to ascertain whether this proposed interface could be similar but from the description it is not (since it doesn't provide any abilities to modify / control the VM which the snap is running inside, only the User VMs).